File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/Button/Functionality/DialogResultPage.vb
Namespace CompanionKit.Controls.Button.Functionality

	Public Class DialogResultPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
            InitializeComponent()

        End Sub

        '' <summary>
        ''' Handles the Click event of the mobjOpenButton control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjOpenButton_Click(sender As Object, e As EventArgs) Handles mobjOpenButton.Click
            'Show form as dialog
            Dim mobjDialogForm As New CustomDialogForm()
            mobjDialogForm.ShowDialog(AddressOf OnCloseForm)
        End Sub

        ''' <summary>
        ''' Called when [close form].
        ''' </summary>
        ''' <param name="sender">The sender.</param>
        ''' <param name="args">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub OnCloseForm(sender As Object, args As EventArgs)
            'Show appropriate message
            Dim objForm As CustomDialogForm = DirectCast(sender, CustomDialogForm)
            If objForm.DialogResult = DialogResult.Yes Then
                MessageBox.Show("'Yes' button has been clicked")
            ElseIf objForm.DialogResult = DialogResult.No Then
                MessageBox.Show("'No' button has been clicked")
            End If
        End Sub

	End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/Button/Functionality/DialogResultPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;


namespace CompanionKit.Controls.Button.Functionality
{
    public partial class DialogResultPage : UserControl
    {
        public DialogResultPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles the Click event of the mobjOpenButton control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjOpenButton_Click(object sender, EventArgs e)
        {
            //Show form as dialog
            CustomDialogForm mobjDialogForm = new CustomDialogForm();
            mobjDialogForm.ShowDialog(OnCloseForm);
        }

        /// <summary>
        /// Called when [close form].
        /// </summary>
        /// <param name="sender">The sender.</param>
        /// <param name="args">The <see cref="EventArgs"/> instance containing the event data.</param>
        void OnCloseForm(object sender, EventArgs args)
        {
            //Show appropriate message
            CustomDialogForm objForm = (CustomDialogForm)sender;
            if (objForm.DialogResult == DialogResult.Yes)
                MessageBox.Show("'Yes' button has been clicked");
            else if (objForm.DialogResult == DialogResult.No)
                MessageBox.Show("'No' button has been clicked");
        }
    }
}